packages/web-components/src/counter-badge/counter-badge.ts

```diff
@@ -0,0 +1,125 @@
+import { attr, FASTElement, nullableNumberConverter } from '@microsoft/fast-element';
+import { applyMixins, StartEnd } from '@microsoft/fast-foundation';
+import {
+  CounterBadgeAppearance,
+  CounterBadgeColor,
+  CounterBadgeShape,
+  CounterBadgeSize,
+} from './counter-badge.options.js';
+
+/**
+ * The base class used for constructing a fluent-badge custom element
+ * @public
+ */
+export class CounterBadge extends FASTElement {
+  /**
+   * The appearance the badge should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: appearance
+   */
+  @attr
+  public appearance: CounterBadgeAppearance;
+
+  /**
+   * The color the badge should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: color
+   */
+  @attr
+  public color: CounterBadgeColor;
+  /**
+   * The shape the badge should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: shape
+   */
+  @attr
+  public shape: CounterBadgeShape;
+
+  /**
+   * The size the badge should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: size
+   */
+  @attr
+  public size: CounterBadgeSize;
+
+  /**
+   * The count the badge should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: count
+   */
+  @attr({ converter: nullableNumberConverter })
+  public count: number = 0;
+  protected countChanged() {
+    this.setCount();
+  }
+
+  /**
+   * Max number to be displayed
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: overflow-count
+   */
+  @attr({ attribute: 'overflow-count', converter: nullableNumberConverter })
+  public overflowCount: number = 99;
```

Comment:
This is my only remaining concern for the PR - there are still attributes with default values reflected back to DOM.
This is a case which cannot be solved by pure CSS and would require additional JS boilerplate :-(